src/prometheus-net.DotNetRuntime.Tests/DotNetRuntimeStatsBuilderTests.cs

```diff
@@ -73,5 +73,27 @@ public void StartCollecting_Allows_A_New_Collector_To_Run_After_Disposing_A_Prev
             {
             }
         }
+
+        [Test]
+        public void StartCollecting_Allows_Multiple_Collectors_For_Non_Default_Registries()
+        {
+#if PROMV2
+            var registry1 = new DefaultCollectorRegistry();
+            var registry2 = new DefaultCollectorRegistry();
+#elif PROMV3
+            var registry1 = Metrics.NewCustomRegistry();
+            var registry2 = Metrics.NewCustomRegistry();
+#endif
+
+            using (DotNetRuntimeStatsBuilder.Customize().StartCollecting())
```

Comment:
Once the changes mentioned below have been made, it would be good to verify that you can't call `StartCollecting` twice on the same custom registry.

Reply:
Okay - test are updated to verify nested registrations fail in all necessary cases, and re-registering after dispose will succeed.